eslint-plugin-prettier.js

```diff
@@ -401,19 +402,22 @@ module.exports = {
               usePrettierrc &&
               prettier.resolveConfig &&
               prettier.resolveConfig.sync
-                ? prettier.resolveConfig.sync(context.getFilename(), {
+                ? prettier.resolveConfig.sync(sourceFileName, {
                     editorconfig: true
                   })
                 : null;
             const prettierOptions = Object.assign(
               {},
               prettierRcOptions,
               eslintPrettierOptions,
-              { filepath: context.getFilename() }
+              { filepath: sourceFileName }
             );
-
             const prettierSource = prettier.format(source, prettierOptions);
-            if (source !== prettierSource) {
+            const sourceFileInfo = prettier.getFileInfo.sync(sourceFileName, {
```

Comment:
Can you add a test for this behavior?

Reply:
Can you check for the existence of `prettier.getFileInfo.sync` before calling it? The plugin needs to continue being able to work for users on old versions of `prettier`.

Reply:
I’ll look into adding a test for this.

As regards checking for the existence of `prettier.getFileInfo.sync` I could do this, but this functionality is required in order to test for whether a file is being ignored,  and so I’ve bumped the dependency to prettier `^1.13.0` for that reason. Is there a reason even future releases of `eslint-plugin-prettier` _needs_ to always support old versions of prettier?

Maybe this be a minor release instead of a patch release, perhaps?

Reply:
Since the user installs `prettier` independently of `eslint-plugin-prettier`, it would be a breaking change to drop support for old versions of `prettier`, because we don't have any control over what version the user has installed.

Reply:
I assume that’s an intentional thing, rather than explicitly having this depend on a given version of prettier? I’m not sure it would be possible to add a test to cover checking for a file being correctly ignored if we can’t guarantee a minimum version of prettier is installed though. Would welcome ideas if you think that is possible!

Reply:
Yes, the reasoning is that the user might want to choose a specific version of `prettier` to lint their code with (also see: https://github.com/prettier/eslint-plugin-prettier/pull/1). Admittedly, this was probably more important at the time that pull request was created than it is now, because prettier wasn't yet stable and was making breaking changes more frequently, and people wanted to use the plugin with several different versions.

Reply:
Cool. Like I say this can’t be merged until #98 is fixed anyway, so I guess I can revisit when that is handled (or maybe dive in and have a go at fixing that myself! 😁)

Reply:
Sure, that would be great.